packages/hyperion-autologging/src/ALSurface.ts

```diff
@@ -282,13 +283,23 @@ export function init(options: InitOptions): ALSurfaceHOC {
       }, [domAttributeName, domAttributeValue]);
     }
 
+    ReactModule.useLayoutEffect(() => {
+      const nodeRefCurrent = props.nodeRef?.current;
+
+      if (nodeRefCurrent == null) {
+        return;
+      }
+      nodeRefCurrent.setAttribute(domAttributeName, domAttributeValue);
+    }, [props.nodeRef]);
+
+
     flowlet.data.surface = surfacePath;
     let children = props.children;
 
     if (!options.disableReactDomPropsExtension) {
       const foundDomElement = propagateFlowletDown(props.children, surfaceData);
 
-      if (foundDomElement !== true) {
+      if (foundDomElement !== true && props.nodeRef == null) {
```

Comment:
Better to add something like `if ref != null` as the first and make these a bunch of `elses` 
Or wrap the whole thing in one if instead of adding the condition to each case.